@wjwwood Here's the output

```
/home/chadrockey/wkspc/src/ros_comm/roscpp/src/libros/timer.cpp: In constructor ‘ros::Timer::Timer(const ros::TimerOptions&)’:
/home/chadrockey/wkspc/src/ros_comm/roscpp/src/libros/timer.cpp:98:30: error: cannot convert ‘const VoidConstPtr {aka const boost::shared_ptr<const void>}’ to ‘bool’ in assignment
   impl_->has_tracked_object_ = ops.tracked_object;
                              ^
/home/chadrockey/wkspc/src/ros_comm/roscpp/src/libros/wall_timer.cpp: In constructor ‘ros::WallTimer::WallTimer(const ros::WallTimerOptions&)’:
/home/chadrockey/wkspc/src/ros_comm/roscpp/src/libros/wall_timer.cpp:98:30: error: cannot convert ‘const VoidConstPtr {aka const boost::shared_ptr<const void>}’ to ‘bool’ in assignment
   impl_->has_tracked_object_ = ops.tracked_object;

```

From my research, c++11 removes some auto casting to boolean (the safe bool idiom).  The fix is to explicity call the type you wish to cast to.  I think this is so you don't do funny things like bitshifting an auto bool cast of a pointer and whatnot.

--- a/clients/roscpp/src/libros/timer.cpp
+++ b/clients/roscpp/src/libros/timer.cpp
@@ -95,7 +95,7 @@ Timer::Timer(const TimerOptions& ops)
   impl_->callback_ = ops.callback;
   impl_->callback_queue_ = ops.callback_queue;
   impl_->tracked_object_ = ops.tracked_object;
-  impl_->has_tracked_object_ = ops.tracked_object;
+  impl_->has_tracked_object_ = (ops.tracked_object != NULL);
   impl_->oneshot_ = ops.oneshot;
 }
 

--- a/clients/roscpp/src/libros/wall_timer.cpp
+++ b/clients/roscpp/src/libros/wall_timer.cpp
@@ -95,7 +95,7 @@ WallTimer::WallTimer(const WallTimerOptions& ops)
   impl_->callback_ = ops.callback;
   impl_->callback_queue_ = ops.callback_queue;
   impl_->tracked_object_ = ops.tracked_object;
-  impl_->has_tracked_object_ = ops.tracked_object;
+  impl_->has_tracked_object_ = (ops.tracked_object != NULL);
   impl_->oneshot_ = ops.oneshot;
 }
 

